File: js/schedule.js
const fs = require('fs');
const path = require('path');

function readSchedule(filename) {
    const schedule = {};
    const data = fs.readFileSync(filename, 'utf8');
    const lines = data.trim().split('\n').slice(1); // Skip header line

    lines.forEach(line => {
        const parts = line.split(',').map(part => part.trim());
        const providerName = parts[0];
        const startTime = parts[1];
        const endTime = parts[2];
        schedule[providerName] = { startTime, endTime };
    });

    return schedule;
}

function schedulingChallenge(filename) {
    const schedule = readSchedule(filename);
    console.log(schedule);
}

const filename = path.join(__dirname, 'test.txt');
schedulingChallenge(filename);

File: python/schedule.py
def read_schedule(filename):
    schedule = {}
    with open(filename, 'r') as file:
        next(file)
        for line in file:
            parts = line.strip().split(',')
            provider_name = parts[0].strip()
            start_time = parts[1].strip()
            end_time = parts[2].strip()
            schedule[provider_name] = (start_time, end_time)
    return schedule

def schedulingChallenge(filename):
    schedule = read_schedule(filename)
    print(schedule)


filename = 'test.txt'
response = schedulingChallenge(filename=filename)